### DISCUSSION TITLE
Beginner questions

### DISCUSSION THREAD

**Discussion by chaoticbob** [donmccurdy/glTF-Transform#1005] [Q&A] [answered]
Hi - I'm just getting started with *glTF-Transform* and had some questions that I'm not able to find in documentation, web searches, or the source code. Was hoping to get some assistance from the community.

1. When using `uastc`, is there a way to specify an output directory for the `*.ktx2` files?
2. One of my files has an ICC profile and `toktx` is saying it needs `--assign_oetf`, is there a way i can force `srgb` texture or maybe all the textures? I looked at the output for `-v` and it looks everything else is working fine except for this one file.

Thanks!

**Comment by donmccurdy** [comment]
Hi @chaoticbob!

> When using uastc, is there a way to specify an output directory for the *.ktx2 files?

Not presently. There's a lot of interest in being able to generate KTX2 files without using the CLI, see https://github.com/donmccurdy/glTF-Transform/issues/675, in which case it would be easier to set output directories in your script. But for the time being KTX2 compression is only available through the CLI.

> ... is there a way i can force srgb texture or maybe all the textures?

I think this would require an `else` block in the section below, assigning "srgb" if the first check falls through.

https://github.com/donmccurdy/glTF-Transform/blob/66dded6547d078b3fceba2d2dc7624bd10e9b2ec/packages/cli/src/transforms/toktx.ts#L313-L316

Is that something you'd want to make a PR for?

**Reply by chaoticbob** [reply]
Thanks @donmccurdy!

I'd be happy to attempt a PR but I need to get my head around node of node first :)

...speaking of which....I'm trying to get a dev environment setup for glTF-Transform by folowing the *Contributing* section of the docs, but I'm running into an error when executing  `yarn run dist` (error log below).

And again, I wasn't able to find anything that could help me on web searches. Am I missing a step or possibly using an incorrect version of npm or node? I'm using `npm 9.6.7` and `node v20.3.1`.


**Error Message**
```
c:\code\source\glTF-Transform>npx yarn run dist
yarn run v1.22.19
$ lerna run dist
lerna notice cli v7.0.2


    ×  @gltf-transform/docs:dist
$ vite build

vite v4.3.9 building SSR bundle for production...
transforming...
✓ 76 modules transformed.
rendering chunks...

       node:internal/event_target:1037
         process.nextTick(() => { throw err; });
                                  ^
       Error: File not found: /c:/code/source/glTF-Transform/packages/core/src/core.ts
           at DirectoryCoordinator.addSourceFileAtPath (c:\code\source\glTF-Transform\node_modules\ts-morph\dist\ts-morph.js:18849:19)
           at Project.addSourceFileAtPath (c:\code\source\glTF-Transform\node_modules\ts-morph\dist\ts-morph.js:20307:51)
           at Parser.addModule (file:///c:/code/source/glTF-Transform/node_modules/@greendoc/parse/dist/greendoc-parse.modern.js:51:28)
           at file:///c:/code/source/glTF-Transform/packages/docs/.svelte-kit/output/server/chunks/index3.js:17:36
           at ModuleJob.run (node:internal/modules/esm/module_job:192:25)
           at async DefaultModuleLoader.import (node:internal/modules/esm/loader:246:24)
           at async analyse (file:///c:/code/source/glTF-Transform/node_modules/@sveltejs/kit/src/core/postbuild/analyse.js:59:16)
           at async MessagePort.<anonymous> (file:///c:/code/source/glTF-Transform/node_modules/@sveltejs/kit/src/utils/fork.js:22:16)
       Emitted 'error' event on Worker instance at:
           at [kOnErrorMessage] (node:internal/worker:326:10)
           at [kOnMessage] (node:internal/worker:337:37)
           at MessagePort.<anonymous> (node:internal/worker:232:57)
           at [nodejs.internal.kHybridDispatch] (node:internal/event_target:762:20)
           at exports.emitMessage (node:internal/per_context/messageport:23:28) {
         path: '/c:/code/source/glTF-Transform/packages/core/src/core.ts',
         code: 'ENOENT'
       }

       Node.js v20.3.1
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

    √  @gltf-transform/core:dist (11s)

 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  Lerna (powered by Nx)   Ran target dist for 6 projects (11s)

    √    1/2 succeeded [0 read from cache]

    ×    1/2 targets failed, including the following:
         - @gltf-transform/docs:dist

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Reply by donmccurdy** [reply, marked as answer]
Ah sorry, the documentation does not build on Windows right now. See https://github.com/donmccurdy/glTF-Transform/pull/959.

If you omit the documentation from the build (see below) does it work?

```bash
yarn run dist --ignore "@gltf-transform/docs"
```

I've been developing with Node.js v18, but Node.js v20 should be fine too.

**Reply by chaoticbob** [reply]
That works! Thank you!

If it's okay with you, I'll mark this as answered and will follow up with the PR once I get a bit more comfortable with node dev. 

```
c:\code\source\glTF-Transform>npx yarn run dist --ignore "@gltf-transform/docs"
yarn run v1.22.19
$ lerna run dist --ignore @gltf-transform/docs
lerna notice cli v7.0.2
lerna notice filter excluding "@gltf-transform/docs"
lerna info filter [ '!@gltf-transform/docs' ]

    √  @gltf-transform/core:dist (11s)
    √  @gltf-transform/extensions:dist (15s)
    √  @gltf-transform/functions:dist (15s)
    √  @gltf-transform/test-utils:dist (4s)
    √  @gltf-transform/cli:dist (5s)

 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  Lerna (powered by Nx)   Successfully ran target dist for 5 projects (46s)

Done in 47.33s.```
